Index: src/main/webapp/js/local/variazioni/consulta.importi.ueb.js
===================================================================
--- src/main/webapp/js/local/variazioni/consulta.importi.ueb.js	(revision 11412)
+++ src/main/webapp/js/local/variazioni/consulta.importi.ueb.js	(revision 11413)
@@ -203,6 +203,8 @@
 
 /* Document Ready */
 $(function() {
+	
+	//window.alert('paginaRicaricata');
     var optionsBase = {
         // Chiamata al termine della creazione della tabella
         "fnDrawCallback" : function () {
@@ -241,4 +243,32 @@
             $('#capitoloNonSelezionato').slideUp();
         }
     });
+    
+    /* ***********    CR- 3597                                                         ************ */
+    var $pulsanteSi = $('#pulsanteSiModaleConfermaNoQuadraturaCassa');
+    var $pulsanteNo = $('#pulsanteNoModaleConfermaNoQuadraturaCassa');
+    var $form = $('#aggiornaVariazioneImportiConUEB');
+    var $modale = $("#modaleConfermaQuadraturaCassa");
+    $modale.data("richiedi-conferma");
+    if($modale.data("richiedi-conferma")){
+    	$modale.modal("show");
+    	$pulsanteSi.removeAttr('disabled');    	 
+    }
+    
+    $pulsanteSi.on('click', function(e){
+    	$('form').append('<input type= "hidden" name ="saltaCheckStanziamentoCassa" value ="true" >');
+    	$modale.data("richiedi-conferma", "");
+    	$("#ERRORI").slideUp();
+    	$('form').submit();
+    	$(e.target).attr('disabled');
+
+    });
+    
+    $pulsanteNo.on('click', function(){
+    	$modale.data("richiedi-conferma", "");
+    	$modale.modal("hide");        	
+    });
+    
+   
+
 });
\ No newline at end of file
Index: src/main/webapp/js/local/variazioni/aggiorna.importi.js
===================================================================
--- src/main/webapp/js/local/variazioni/aggiorna.importi.js	(revision 11412)
+++ src/main/webapp/js/local/variazioni/aggiorna.importi.js	(revision 11413)
@@ -882,10 +882,35 @@
             $('#aggiornaVariazioneImporti').submit();
         });
 
+        
+        /* ***********    CR- 3597                                                         ************ */
+        var $pulsanteSi = $('#pulsanteSiModaleConfermaNoQuadraturaCassa');
+        var $pulsanteNo = $('#pulsanteNoModaleConfermaNoQuadraturaCassa');
+        var $form = $('#aggiornaVariazioneImportiConUEB');
+        var $modale = $("#modaleConfermaQuadraturaCassa");
+        $modale.data("richiedi-conferma");
+        if($modale.data("richiedi-conferma")){
+        	$modale.modal("show");
+        	$pulsanteSi.removeAttr('disabled');
+        }
+        
+        $pulsanteSi.on('click', function(e){
+        	$('form').append('<input type= "hidden" name ="saltaCheckStanziamentoCassa" value ="true" >');
+        	$(e.target).attr('disabled');
+        	$('form').submit();
+        	$modale.data("richiedi-conferma", "");
+        });
+        
+        $pulsanteNo.on('click', function(){
+        	$modale.data("richiedi-conferma", "");
+        	$modale.modal("hide");        	
+        });
+        
         //if($('#annoProvvedimento').val() !== ""){
         //   $('#SPAN_InformazioniProvvedimento').html(": " + $('#annoProvvedimento').val() + " / " + $('#numeroProvvedimento').val());
         //}
-
+        
+              
         $("form").on("reset", function() {
             // NATIVO
             this.reset();
Index: src/main/webapp/js/local/variazioni/consulta.importi.js
===================================================================
--- src/main/webapp/js/local/variazioni/consulta.importi.js	(revision 11412)
+++ src/main/webapp/js/local/variazioni/consulta.importi.js	(revision 11413)
@@ -66,5 +66,32 @@
     var options = $.extend(true, {}, ConsultaVariazioni.dataTableBaseOptions,optionsBase);
 
     $("#elencoCapitoli").dataTable(options);
+    
+    var $pulsanteSi = $('#pulsanteSiModaleConfermaNoQuadraturaCassa');
+    var $pulsanteNo = $('#pulsanteNoModaleConfermaNoQuadraturaCassa');
+    var $form = $('#aggiornaVariazioneImportiConUEB');
+    var $modale = $("#modaleConfermaQuadraturaCassa");
+    $modale.data("richiedi-conferma");
+    if($modale.data("richiedi-conferma")){
+    	$modale.modal("show");
+    	$pulsanteSi.removeAttr('disabled');    	
+    }
+    
+    $pulsanteSi.on('click', function(){
+    	$('form').append('<input type= "hidden" name ="saltaCheckStanziamentoCassa" value ="true" >');
+    	$(e.target).attr('disabled');
+    	$('form').submit();
+    	$modale.data("richiedi-conferma", "");
+    	$('#ERRORI').slideUp();
+    });
+	
+	$pulsanteNo.on('click', function(){
+    	$modale.data("richiedi-conferma", "");
+    	$modale.modal("hide");        	
+    });
+    
+    
+    
+    
 
 });
\ No newline at end of file
Index: src/main/webapp/js/local/variazioni/aggiorna.importi.ueb.js
===================================================================
--- src/main/webapp/js/local/variazioni/aggiorna.importi.ueb.js	(revision 11412)
+++ src/main/webapp/js/local/variazioni/aggiorna.importi.ueb.js	(revision 11413)
@@ -1323,6 +1323,32 @@
 //            $('#SPAN_InformazioniProvvedimento').html(": " + $('#annoProvvedimento').val() + " / " + $('#numeroProvvedimento').val());
 //        }
 
+/* ***********    CR- 3597                                                         ************ */
+        var $pulsanteSi = $('#pulsanteSiModaleConfermaNoQuadraturaCassa');
+        var $pulsanteNo = $('#pulsanteNoModaleConfermaNoQuadraturaCassa');
+        var $form = $('#aggiornaVariazioneImportiConUEB');
+        var $modale = $("#modaleConfermaQuadraturaCassa");
+        $modale.data("richiedi-conferma");
+        if($modale.data("richiedi-conferma")){
+        	$modale.modal("show");
+        	$pulsanteSi.removeAttr('disabled');
+        }
+        
+        $pulsanteSi.on('click', function(e){
+        	$('form').append('<input type= "hidden" name ="saltaCheckStanziamentoCassa" value ="true" >');
+        	$('form').submit();
+        	$(e.target).attr('disabled');
+        	$modale.data("richiedi-conferma", "");
+        	
+        });
+        
+        $pulsanteNo.on('click', function(){
+        	$modale.data("richiedi-conferma", "");
+        	$modale.modal("hide");        	
+        });
+        
+
+        
         $("form").on("reset", function() {
             // NATIVO
             this.reset();
Index: src/main/webapp/jsp/variazione/aggiornaVariazioneImportiUEB.jsp
===================================================================
--- src/main/webapp/jsp/variazione/aggiornaVariazioneImportiUEB.jsp	(revision 11412)
+++ src/main/webapp/jsp/variazione/aggiornaVariazioneImportiUEB.jsp	(revision 11413)
@@ -263,6 +263,19 @@
 							</div>
 						</div>
 					</div>
+					<div id="modaleConfermaQuadraturaCassa" class="modal hide fade" tabindex="-1" role="dialog" <s:if test ="%{richiediConfermaQuadratura}"> data-richiedi-conferma="richiedi" </s:if> aria-labelledby="labelModaleConfermaNoProvvedimento" aria-hidden="true">
+		<div class="modal-body" id="labelModaleConfermaQuadraturaCassa">
+			<div class="alert alert-warning alert-persistent">
+				<button data-dismiss="modal" class="close" type="button">&times;</button>
+				<p><strong>Attenzione!</strong></p>
+				<p>Non vi &eacute; quadratura sulla cassa. Proseguire ugualmente con il salvataggio?</p>
+			</div>
+		</div>
+		<div class="modal-footer">
+			<button id = "pulsanteNoModaleConfermaNoQuadraturaCassa" type="button" data-dismiss="modal" class="btn">no, indietro</button>
+			<button type="button" class="btn btn-primary" id="pulsanteSiModaleConfermaNoQuadraturaCassa">si, prosegui</button>
+		</div>
+	</div>
 				</s:form>
 			</div>
 		</div>
Index: src/main/webapp/jsp/variazione/modaleConfermaQuadraturaCassa.jsp
===================================================================
--- src/main/webapp/jsp/variazione/modaleConfermaQuadraturaCassa.jsp	(revision 0)
+++ src/main/webapp/jsp/variazione/modaleConfermaQuadraturaCassa.jsp	(revision 11413)
@@ -0,0 +1,14 @@
+<%@ page language="java" contentType="text/html; charset=UTF-8" pageEncoding="UTF-8"%>
+<div id="modaleConfermaQuadraturaCassa" class="modal hide fade" tabindex="-1" role="dialog" aria-labelledby="labelModaleConfermaNoProvvedimento" aria-hidden="true">
+	<div class="modal-body" id="labelModaleConfermaQuadraturaCassa">
+		<div class="alert alert-warning alert-persistent">
+			<button data-dismiss="modal" class="close" type="button">&times;</button>
+			<p><strong>Attenzione!</strong></p>
+			<p>Non vi &eacute; quadratura sulla cassa. Proseguire ugualmente con il salvataggio?</p>
+		</div>
+	</div>
+	<div class="modal-footer">
+		<button id = "pulsanteNoModaleConfermaNoQuadraturaCassa" type="button" data-dismiss="modal" class="btn">no, indietro</button>
+		<button type="button" class="btn btn-primary" id="pulsanteSiModaleConfermaNoQuadraturaCassa">si, prosegui</button>
+	</div>
+</div>

Property changes on: src/main/webapp/jsp/variazione/modaleConfermaQuadraturaCassa.jsp
___________________________________________________________________
Added: svn:mime-type
## -0,0 +1 ##
+text/plain
\ No newline at end of property
Index: src/main/webapp/jsp/variazione/definisceVariazioneImportiUEB.jsp
===================================================================
--- src/main/webapp/jsp/variazione/definisceVariazioneImportiUEB.jsp	(revision 11412)
+++ src/main/webapp/jsp/variazione/definisceVariazioneImportiUEB.jsp	(revision 11413)
@@ -133,6 +133,20 @@
 								<s:submit cssClass="btn btn-primary pull-right" value="definisci variazione" />
 							</s:if>
 						</p>
+						<!-- CR-3597 -->
+						<div id="modaleConfermaQuadraturaCassa" class="modal hide fade" tabindex="-1" role="dialog" <s:if test ="%{richiediConfermaQuadratura}"> data-richiedi-conferma="richiedi" </s:if> aria-labelledby="labelModaleConfermaNoProvvedimento" aria-hidden="true">
+							<div class="modal-body" id="labelModaleConfermaQuadraturaCassa">
+								<div class="alert alert-warning alert-persistent">
+									<button data-dismiss="modal" class="close" type="button">&times;</button>
+									<p><strong>Attenzione!</strong></p>
+									<p>Non vi &eacute; quadratura sulla cassa. Proseguire ugualmente con il salvataggio?</p>
+								</div>
+							</div>
+							<div class="modal-footer">
+								<button id = "pulsanteNoModaleConfermaNoQuadraturaCassa" type="button" data-dismiss="modal" class="btn">no, indietro</button>
+								<button type="button" class="btn btn-primary" id="pulsanteSiModaleConfermaNoQuadraturaCassa">si, prosegui</button>
+							</div>
+						</div>						
 					</s:form>
 				</div>
 			</div>
Index: src/main/webapp/jsp/variazione/aggiornaVariazioneImporti.jsp
===================================================================
--- src/main/webapp/jsp/variazione/aggiornaVariazioneImporti.jsp	(revision 11412)
+++ src/main/webapp/jsp/variazione/aggiornaVariazioneImporti.jsp	(revision 11413)
@@ -13,7 +13,7 @@
 			<div class="span12 contentPage">
 				<s:include value="/jsp/include/messaggi.jsp" />
 				<s:form cssClass="form-horizontal" action="concludiAggiornamentoVariazioneImporti" novalidate="novalidate" id="aggiornaVariazioneImporti" method="post">
-					<h3>Aggiorna Variazione</h3>
+					<h3>Aggiorna Variazione </h3>					
 					<div class="accordion-group">
 						<div class="accordion-heading">
 							<a class="accordion-toggle" href="#collapseVariazioni" data-parent="#accordion2" data-toggle="collapse">
@@ -21,6 +21,7 @@
 							</a>
 						</div>
 						<div id="collapseVariazioni" class="accordion-body in collapse" style="height: auto;">
+							<s:hidden id="hidden_ignoraQuadraturaDiCassa" name ="ignoraQuadraturaDiCassa2" value =""/>
 							<div class="accordion-inner">
 								<dl class="dl-horizontal">
 									<dt>Num. variazione</dt>
@@ -230,6 +231,19 @@
 							</div>
 						</div>
 					</div>
+					<div id="modaleConfermaQuadraturaCassa" class="modal hide fade" tabindex="-1" role="dialog" <s:if test ="%{richiediConfermaQuadratura}"> data-richiedi-conferma="richiedi" </s:if> aria-labelledby="labelModaleConfermaNoProvvedimento" aria-hidden="true">
+						<div class="modal-body" id="labelModaleConfermaQuadraturaCassa">
+							<div class="alert alert-warning alert-persistent">
+								<button data-dismiss="modal" class="close" type="button">&times;</button>
+								<p><strong>Attenzione!</strong></p>
+								<p>Non vi &eacute; quadratura sulla cassa. Proseguire ugualmente con il salvataggio?</p>
+							</div>
+						</div>
+						<div class="modal-footer">
+							<button id = "pulsanteNoModaleConfermaNoQuadraturaCassa" type="button" data-dismiss="modal" class="btn">no, indietro</button>
+							<button type="button" class="btn btn-primary" id="pulsanteSiModaleConfermaNoQuadraturaCassa">si, prosegui</button>
+						</div>
+					</div>
 				</s:form>
 			</div>
 		</div>
@@ -311,6 +325,8 @@
 	</div>
 	<%-- /Modale annulla --%>
 	
+	
+	
 	<s:include value="/jsp/include/modaleConfermaProsecuzioneSuAzione.jsp" />
 	<s:include value="/jsp/include/footer.jsp"/>
 	<s:include value="/jsp/include/javascript.jsp" />
Index: src/main/webapp/jsp/variazione/definisceVariazioneImporti.jsp
===================================================================
--- src/main/webapp/jsp/variazione/definisceVariazioneImporti.jsp	(revision 11412)
+++ src/main/webapp/jsp/variazione/definisceVariazioneImporti.jsp	(revision 11413)
@@ -95,6 +95,20 @@
 								<s:submit cssClass="btn btn-primary pull-right" value="definisci variazione" />
 							</s:if>
 						</p>
+						<!-- CR-3597 -->
+						<div id="modaleConfermaQuadraturaCassa" class="modal hide fade" tabindex="-1" role="dialog" <s:if test ="%{richiediConfermaQuadratura}"> data-richiedi-conferma="richiedi" </s:if> aria-labelledby="labelModaleConfermaNoProvvedimento" aria-hidden="true">
+							<div class="modal-body" id="labelModaleConfermaQuadraturaCassa">
+								<div class="alert alert-warning alert-persistent">
+									<button data-dismiss="modal" class="close" type="button">&times;</button>
+									<p><strong>Attenzione!</strong></p>
+									<p>Non vi &eacute; quadratura sulla cassa. Proseguire ugualmente con il salvataggio?</p>
+								</div>
+							</div>
+							<div class="modal-footer">
+								<button id = "pulsanteNoModaleConfermaNoQuadraturaCassa" type="button" data-dismiss="modal" class="btn">no, indietro</button>
+								<button type="button" class="btn btn-primary" id="pulsanteSiModaleConfermaNoQuadraturaCassa">si, prosegui</button>
+							</div>
+						</div>	
 					</s:form>
 				</div>
 			</div>
Index: src/main/java/it/csi/siac/siacbilapp/frontend/ui/action/variazione/AggiornaVariazioneImportiAction.java
===================================================================
--- src/main/java/it/csi/siac/siacbilapp/frontend/ui/action/variazione/AggiornaVariazioneImportiAction.java	(revision 11412)
+++ src/main/java/it/csi/siac/siacbilapp/frontend/ui/action/variazione/AggiornaVariazioneImportiAction.java	(revision 11413)
@@ -40,6 +40,7 @@
 import it.csi.siac.siaccorser.model.Informazione;
 import it.csi.siac.siaccorser.model.ServiceResponse;
 import it.csi.siac.siaccorser.model.errore.ErroreCore;
+import it.csi.siac.siacfin2ser.model.errore.ErroreFin;
 
 /**
  * Classe per la gestione dell'aggiornamento della variazione degli importi.
@@ -80,6 +81,7 @@
 			sessionHandler.setParametro(BilSessionParameter.INSERIMENTO_NUOVA_UEB, null);
 		}
 
+		//model.setRichiediConfermaQuadratura(Boolean.FALSE);
 		log.debugEnd(methodName, "");
 	}
 
@@ -176,7 +178,7 @@
 
 		if (response.hasErrori()) {
 			log.debug(methodName, "Invocazione terminata con fallimento");
-			addErrori(response);
+			addErrori(response);			
 			return createSuffix(INPUT);
 		}
 
@@ -273,6 +275,10 @@
 		if (response.hasErrori()) {
 			log.debug(methodName, "Invocazione terminata con fallimento");
 			addErrori(response);
+			
+			Boolean richiediConfermaQuadraturaCassa = response.verificatoErrore(ErroreBil.QUADRATURA_NON_CORRETTA.getCodice()) && Boolean.TRUE.equals(response.getIsQuadraturaCorrettaStanziamento()) && !Boolean.TRUE.equals(response.getIsQuadraturaCorrettaStanziamentoCassa()); 
+			model.setRichiediConfermaQuadratura(richiediConfermaQuadraturaCassa);
+			
 			return createSuffix(INPUT);
 		}
 
Index: src/main/java/it/csi/siac/siacbilapp/frontend/ui/action/variazione/DefinisceVariazioneImportiAction.java
===================================================================
--- src/main/java/it/csi/siac/siacbilapp/frontend/ui/action/variazione/DefinisceVariazioneImportiAction.java	(revision 11412)
+++ src/main/java/it/csi/siac/siacbilapp/frontend/ui/action/variazione/DefinisceVariazioneImportiAction.java	(revision 11413)
@@ -123,6 +123,7 @@
 	 * @return la stringa corrispondente al risultato dell'invocazione
 	 */
 	public String definisciVariazione() {
+		cleanErrori();
 		final String methodName = "definisciVariazione";
 		final String suffisso = "_" + (model.isGestioneUEB() ? "UEB" : "noUEB");
 		
@@ -148,7 +149,13 @@
 	
 		if(response.hasErrori()) {
 			log.debug(methodName, "Errore nella risposta del servizio");
+			
 			addErrori(methodName, response);
+			//
+			Boolean richiediConfermaQuadraturaCassa = response.verificatoErrore(ErroreBil.QUADRATURA_NON_CORRETTA.getCodice()) && Boolean.TRUE.equals(response.getIsQuadraturaCorrettaStanziamento()) && !Boolean.TRUE.equals(response.getIsQuadraturaCorrettaStanziamentoCassa()); 
+			log.debug(methodName, "Richiedo la conferma della squadratura di cassa? " + richiediConfermaQuadraturaCassa);
+			log.debug(methodName, "Nel servizio ho avuto l'errore sulla quadratura? " + response.verificatoErrore(ErroreBil.QUADRATURA_NON_CORRETTA.getCodice()));
+			model.setRichiediConfermaQuadratura(richiediConfermaQuadraturaCassa);
 			return INPUT + suffisso;
 		}
 
@@ -157,7 +164,8 @@
 		
 		model.setDefinizioneEseguita(Boolean.TRUE);
 		model.setStatoVariazione(StatoOperativoVariazioneDiBilancio.DEFINITIVA);
-		
+		//non si e' verificato l'errore: non devo richiedere la conferma
+		model.setRichiediConfermaQuadratura(Boolean.FALSE);
 		// Imposto i dati nel model
 		model.injettaDatiPostDefinizione(response);
 		return SUCCESS + suffisso;
Index: src/main/java/it/csi/siac/siacbilapp/frontend/ui/model/variazione/DefinisceVariazioneImportiModel.java
===================================================================
--- src/main/java/it/csi/siac/siacbilapp/frontend/ui/model/variazione/DefinisceVariazioneImportiModel.java	(revision 11412)
+++ src/main/java/it/csi/siac/siacbilapp/frontend/ui/model/variazione/DefinisceVariazioneImportiModel.java	(revision 11413)
@@ -58,6 +58,9 @@
 	private StatoOperativoVariazioneDiBilancio statoVariazione;
 	private String noteVariazione;
 	
+	private Boolean richiediConfermaQuadratura = Boolean.FALSE;
+	private String saltaCheckStanziamentoCassa = "";
+	
 	private Integer uidProvvedimento;
 	private String tipoProvvedimento;
 	private Integer annoProvvedimento;
@@ -763,6 +766,7 @@
 		request.setRichiedente(getRichiedente());
 		request.setIdAttivita(idAttivita);
 		request.setVariazioneImportoCapitolo(variazioneImportoCapitolo);
+		request.setSaltaCheckStanziamentoCassa(Boolean.parseBoolean(saltaCheckStanziamentoCassa));
 		request.setListaVariabiliProcesso(getAzioneRichiesta().getVariabiliProcesso());
 		
 		return request;
@@ -865,5 +869,29 @@
 		differenzaResiduo = importiVariazione.getDifferenzaResiduo();
 		differenzaFondoPluriennaleVincolato = importiVariazione.getDifferenzaFondoPluriennaleVincolato();
 	}
+
+
+
+	public Boolean getRichiediConfermaQuadratura() {
+		return richiediConfermaQuadratura;
+	}
+
+
+
+	public void setRichiediConfermaQuadratura(Boolean richiediConfermaQuadratura) {
+		this.richiediConfermaQuadratura = richiediConfermaQuadratura;
+	}
+
+
+
+	public String getSaltaCheckStanziamentoCassa() {
+		return saltaCheckStanziamentoCassa;
+	}
+
+
+
+	public void setSaltaCheckStanziamentoCassa(String saltaCheckStanziamentoCassa) {
+		this.saltaCheckStanziamentoCassa = saltaCheckStanziamentoCassa;
+	}
 	
 }
Index: src/main/java/it/csi/siac/siacbilapp/frontend/ui/model/variazione/AggiornaVariazioneModel.java
===================================================================
--- src/main/java/it/csi/siac/siacbilapp/frontend/ui/model/variazione/AggiornaVariazioneModel.java	(revision 11412)
+++ src/main/java/it/csi/siac/siacbilapp/frontend/ui/model/variazione/AggiornaVariazioneModel.java	(revision 11413)
@@ -93,9 +93,14 @@
 	private Boolean concludiAbilitato = Boolean.TRUE;
 	private Boolean campiNonModificabili = Boolean.FALSE;
 	
+	private Boolean richiediConfermaQuadratura = Boolean.FALSE;
+	private String saltaCheckStanziamentoCassa = "";
+	
 	// Per cachare le responses
 	private Map<String, String> cache = new HashMap<String, String>();
+
 	
+	
 	/** Costruttore vuoto di default */
 	public AggiornaVariazioneModel() {
 		specificaImporti = new SpecificaVariazioneImportoModel();
@@ -720,8 +725,10 @@
 		request.setIdAttivita(idAttivita);
 		request.setInvioOrganoAmministrativo(invioAdOrganoAmministrativo);
 		request.setInvioOrganoLegislativo(invioAdOrganoLegislativo);
-		request.setRichiedente(getRichiedente());
+        request.setSaltaCheckStanziamentoCassa(Boolean.parseBoolean(saltaCheckStanziamentoCassa));
 		
+        request.setRichiedente(getRichiedente());
+		
 		request.setVariazioneImportoCapitolo(creaUtilityVariazioneImportoCapitolo());
 		
 		return request;
@@ -979,4 +986,26 @@
 		setStringaProvvedimento(stringa);
 	}
 
+	public Boolean getRichiediConfermaQuadratura() {
+		return richiediConfermaQuadratura;
+	}
+
+	public void setRichiediConfermaQuadratura(Boolean richiediConfermaQuadratura) {
+		this.richiediConfermaQuadratura = richiediConfermaQuadratura;
+	}
+
+	/**
+	 * @return the saltaCheckStanziamentoCassa
+	 */
+	public String getSaltaCheckStanziamentoCassa() {
+		return saltaCheckStanziamentoCassa;
+	}
+
+	/**
+	 * @param saltaCheckStanziamentoCassa the saltaCheckStanziamentoCassa to set
+	 */
+	public void setSaltaCheckStanziamentoCassa(String saltaCheckStanziamentoCassa) {
+		this.saltaCheckStanziamentoCassa = saltaCheckStanziamentoCassa;
+	}
+	
 }
