diff --git src/main/java/it/csi/siac/siacbasegengsaapp/frontend/ui/action/primanotaintegrata/RicercaPrimaNotaIntegrataBaseAction.java src/main/java/it/csi/siac/siacbasegengsaapp/frontend/ui/action/primanotaintegrata/RicercaPrimaNotaIntegrataBaseAction.java
index 85a5d92..7bca3db 100644
--- src/main/java/it/csi/siac/siacbasegengsaapp/frontend/ui/action/primanotaintegrata/RicercaPrimaNotaIntegrataBaseAction.java
+++ src/main/java/it/csi/siac/siacbasegengsaapp/frontend/ui/action/primanotaintegrata/RicercaPrimaNotaIntegrataBaseAction.java
@@ -213,6 +213,7 @@
 		// Impostazione dati in sessione
 		sessionHandler.setParametro(getBilSessionParameterRisultati(), response.getPrimeNote());
 		sessionHandler.setParametroXmlType(getBilSessionParameterRequest(), request);
+		sessionHandler.setParametro(BilSessionParameter.RIENTRO_POSIZIONE_START, null);
 		return SUCCESS;
 	}
 	
diff --git src/main/java/it/csi/siac/siacbasegengsaapp/frontend/ui/action/primanotalibera/RicercaPrimaNotaLiberaBaseAction.java src/main/java/it/csi/siac/siacbasegengsaapp/frontend/ui/action/primanotalibera/RicercaPrimaNotaLiberaBaseAction.java
index f3270d2..c334538 100644
--- src/main/java/it/csi/siac/siacbasegengsaapp/frontend/ui/action/primanotalibera/RicercaPrimaNotaLiberaBaseAction.java
+++ src/main/java/it/csi/siac/siacbasegengsaapp/frontend/ui/action/primanotalibera/RicercaPrimaNotaLiberaBaseAction.java
@@ -367,5 +367,6 @@
 
 		sessionHandler.setParametro(bilSessionParameterRisultati, response.getPrimeNote());
 		sessionHandler.setParametroXmlType(bilSessionParameterRequest, request);
+		sessionHandler.setParametro(BilSessionParameter.RIENTRO_POSIZIONE_START, null);
 	}
 }
diff --git src/main/java/it/csi/siac/siaccecapp/frontend/ui/action/cassaeconomale/stampe/RicercaStampeCassaEconomaleAction.java src/main/java/it/csi/siac/siaccecapp/frontend/ui/action/cassaeconomale/stampe/RicercaStampeCassaEconomaleAction.java
index 446e485..6a40819 100644
--- src/main/java/it/csi/siac/siaccecapp/frontend/ui/action/cassaeconomale/stampe/RicercaStampeCassaEconomaleAction.java
+++ src/main/java/it/csi/siac/siaccecapp/frontend/ui/action/cassaeconomale/stampe/RicercaStampeCassaEconomaleAction.java
@@ -84,6 +84,7 @@
 
 		log.debug(methodName, "Imposto in sessione la lista");
 		sessionHandler.setParametro(BilSessionParameter.RISULTATI_RICERCA_STAMPE_CEC, response.getListaStampe());
+		sessionHandler.setParametro(BilSessionParameter.RIENTRO_POSIZIONE_START, null);
 		return SUCCESS;
 	}
 	
diff --git src/main/java/it/csi/siac/siacfin2app/frontend/ui/action/allegatoatto/stampe/RicercaStampaAllegatoAttoAction.java src/main/java/it/csi/siac/siacfin2app/frontend/ui/action/allegatoatto/stampe/RicercaStampaAllegatoAttoAction.java
index 9125527..8396458 100644
--- src/main/java/it/csi/siac/siacfin2app/frontend/ui/action/allegatoatto/stampe/RicercaStampaAllegatoAttoAction.java
+++ src/main/java/it/csi/siac/siacfin2app/frontend/ui/action/allegatoatto/stampe/RicercaStampaAllegatoAttoAction.java
@@ -100,6 +100,7 @@
 
 		log.debug(methodName, "Imposto in sessione la lista");
 		sessionHandler.setParametro(BilSessionParameter.RISULTATI_RICERCA_STAMPA_ALLEGATO_ATTO, response.getListaAllegatoAttoStampa());
+		sessionHandler.setParametro(BilSessionParameter.RIENTRO_POSIZIONE_START, null);
 		return SUCCESS;
 	}
 	
diff --git src/main/java/it/csi/siac/siacfin2app/frontend/ui/action/attivitaiva/stampa/RicercaStampaIvaAction.java src/main/java/it/csi/siac/siacfin2app/frontend/ui/action/attivitaiva/stampa/RicercaStampaIvaAction.java
index a0a74c6..c962f96 100644
--- src/main/java/it/csi/siac/siacfin2app/frontend/ui/action/attivitaiva/stampa/RicercaStampaIvaAction.java
+++ src/main/java/it/csi/siac/siacfin2app/frontend/ui/action/attivitaiva/stampa/RicercaStampaIvaAction.java
@@ -88,6 +88,7 @@
 
 		log.debug(methodName, "Imposto in sessione la lista");
 		sessionHandler.setParametro(BilSessionParameter.RISULTATI_RICERCA_STAMPA_IVA, response.getStampeIva());
+		sessionHandler.setParametro(BilSessionParameter.RIENTRO_POSIZIONE_START, null);
 		return SUCCESS;
 	}
 	
diff --git src/main/java/it/csi/siac/siacfin2app/frontend/ui/action/provvisoriocassa/RicercaProvvisorioDiCassaAction.java src/main/java/it/csi/siac/siacfin2app/frontend/ui/action/provvisoriocassa/RicercaProvvisorioDiCassaAction.java
index 4fc09fb..e2f4789 100644
--- src/main/java/it/csi/siac/siacfin2app/frontend/ui/action/provvisoriocassa/RicercaProvvisorioDiCassaAction.java
+++ src/main/java/it/csi/siac/siacfin2app/frontend/ui/action/provvisoriocassa/RicercaProvvisorioDiCassaAction.java
@@ -66,6 +66,7 @@
 		// XXX: purtroppo, la lista restituida dal servizio non e' paginata. Per avere i dati di cui necessito, devo impostare qualche dato in piu' in sessione
 		sessionHandler.setParametro(BilSessionParameter.NUMERO_PAGINA_SERVIZI_FIN, response.getNumPagina());
 		sessionHandler.setParametro(BilSessionParameter.NUMERO_RISULTATI_SERVIZI_FIN, response.getNumRisultati());
+		sessionHandler.setParametro(BilSessionParameter.RIENTRO_POSIZIONE_START, null);
 		return SUCCESS;
 	}
 	
diff --git src/main/java/it/csi/siac/siacgenapp/frontend/ui/action/conti/RicercaPianoDeiContiFINAction.java src/main/java/it/csi/siac/siacgenapp/frontend/ui/action/conti/RicercaPianoDeiContiFINAction.java
index d263692..64a7ebb 100644
--- src/main/java/it/csi/siac/siacgenapp/frontend/ui/action/conti/RicercaPianoDeiContiFINAction.java
+++ src/main/java/it/csi/siac/siacgenapp/frontend/ui/action/conti/RicercaPianoDeiContiFINAction.java
@@ -128,6 +128,7 @@
 		sessionHandler.setParametro(BilSessionParameter.CONTO_DA_PULIRE, Boolean.FALSE);
 		sessionHandler.setParametroXmlType(BilSessionParameter.REQUEST_RICERCA_CONTO, reqRSCF);
 		sessionHandler.setParametro(BilSessionParameter.RISULTATI_RICERCA_FIGLI_CONTO, resRSCF.getContiFiglio());
+		sessionHandler.setParametro(BilSessionParameter.RIENTRO_POSIZIONE_START, null);
 		
 		return SUCCESS;
 	}
diff --git src/main/java/it/csi/siac/siacgenapp/frontend/ui/action/primanotaintegrata/RicercaValidazionePrimaNotaIntegrataFINAction.java src/main/java/it/csi/siac/siacgenapp/frontend/ui/action/primanotaintegrata/RicercaValidazionePrimaNotaIntegrataFINAction.java
index b28b542..d703960 100644
--- src/main/java/it/csi/siac/siacgenapp/frontend/ui/action/primanotaintegrata/RicercaValidazionePrimaNotaIntegrataFINAction.java
+++ src/main/java/it/csi/siac/siacgenapp/frontend/ui/action/primanotaintegrata/RicercaValidazionePrimaNotaIntegrataFINAction.java
@@ -340,6 +340,7 @@
 		// Impostazione dati in sessione
 		sessionHandler.setParametro(BilSessionParameter.RISULTATI_RICERCA_PRIMANOTAINTEGRATA_VALIDABILE_GEN, response.getPrimeNote());
 		sessionHandler.setParametroXmlType(BilSessionParameter.REQUEST_RICERCA_PRIMANOTAINTEGRATA_VALIDABILE_GEN, request);
+		sessionHandler.setParametro(BilSessionParameter.RIENTRO_POSIZIONE_START, null);
 		return SUCCESS;
 	}
 	
diff --git src/main/java/it/csi/siac/siacgsaapp/frontend/ui/action/conti/RicercaPianoDeiContiGSAAction.java src/main/java/it/csi/siac/siacgsaapp/frontend/ui/action/conti/RicercaPianoDeiContiGSAAction.java
index f358be7..f22d070 100644
--- src/main/java/it/csi/siac/siacgsaapp/frontend/ui/action/conti/RicercaPianoDeiContiGSAAction.java
+++ src/main/java/it/csi/siac/siacgsaapp/frontend/ui/action/conti/RicercaPianoDeiContiGSAAction.java
@@ -127,6 +127,7 @@
 		sessionHandler.setParametro(BilSessionParameter.CONTO_DA_PULIRE, Boolean.FALSE);
 		sessionHandler.setParametroXmlType(BilSessionParameter.REQUEST_RICERCA_CONTO, reqRSCF);
 		sessionHandler.setParametro(BilSessionParameter.RISULTATI_RICERCA_FIGLI_CONTO, resRSCF.getContiFiglio());
+		sessionHandler.setParametro(BilSessionParameter.RIENTRO_POSIZIONE_START, null);
 		
 		return SUCCESS;
 	}
